lib/graphiti/util/attribute_check.rb

```diff
@@ -65,7 +65,10 @@ def supported?
       end
 
       def attribute
-        @attribute ||= resource.all_attributes[name]
+        @attribute ||= resource.all_attributes.fetch(name) do |attr|
+          resource.attribute_missing(attr)
```

Comment:
I have no real problem with this code, but what do you think about moving to [this line](https://github.com/graphiti-api/graphiti/blob/e3d4136e1c2b96a9f60d36d8bc41e7c0b9e09acb/lib/graphiti/util/attribute_check.rb#L32). The thought being the default logic of `attribute_missing` should hold the code that raises the error. I can see how that might be a PITA with this as a separate class, just want to get your thoughts.

Reply:
I reworked this class a bit. It now has series of checks one that's deals with attributes, `attribute_check!` that also calls `Resource#attribute_missing`. I didn't want to put any logic in that template method because that would require user to reimplement it or call to `super` all of which break the open closed principle.